src/App.js

```diff
@@ -3,17 +3,56 @@ import GridLayout from "./components/GridLayout";
 import FullLayout from "./components/FullLayout";
 import { connect } from "react-redux";
 import { Route, withRouter } from "react-router-dom";
+import { getConfiguredNow } from "./reducers";
 
-import { fetchAlarms } from "./actions";
+import { fetchAlarms, setDateTimeAction } from "./actions";
 import styles from "./App.css";
 
 class App extends Component {
+  componentWillMount() {
+    this.setDateTimeActionWithNow();
+    const that = this;
+    // update redux time every minute (60000 miliseconds) because redux only saves time on the minute accurate
+    setInterval(function() {
+      that.setDateTimeActionWithNow();
+    }, 60000);
+  }
+
   componentDidMount() {
     if (!this.props.iframeModeActive) {
       this.props.fetchAlarms();
     }
   }
 
+  setDateTimeActionWithNow() {
+    const jsDateObject = new Date();
+    const utcMinutes = jsDateObject.getUTCMinutes();
+    const utcMinutesZeroPrefixed =
+      utcMinutes < 10 ? "0" + utcMinutes : utcMinutes;
+
+    const dateStr =
+      jsDateObject.getUTCFullYear() +
+      "-" +
+      (jsDateObject.getUTCMonth() + 1) +
+      "-" +
+      jsDateObject.getUTCDate();
+    const timeStr = jsDateObject.getUTCHours() + ":" + utcMinutesZeroPrefixed;
+
+    this.props.setDateTimeAction(dateStr, timeStr);
+  }
```

Comment:
Deze logica (die een jsDateObject omzet naar een string afgerond op minuten) zou beter op de reducer kunnen denk ik. Het is een constraint van de onze data store die daar bewaakt moet worden.

Reply:
Ja denk dat je gelijk hebt. Zolang de Date dan maar als parameter wordt meegegeven.
De regel "const jsDateObject = new Date();" moet dan uit deze functie. Dit om te voorkomen dat reducer statefull wordt.